### PR TITLE
[Merged by Bors] - feat: port Logic/Relator

### PR DESCRIPTION
**Pull request by dwrensha**
Port logic/relator.lean and add comments in the places where the linter demands them.

Compare to the mathlib3 version: https://github.com/leanprover-community/mathlib/blob/7362d50a80c3e7f7a1fb5adb1c0b6ba593db103d/src/logic/relator.lean

### CODE REVIEW

File: Mathlib/Logic/Relator.lean [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+variable {α : Sort u₁} {β : Sort u₂} {γ : Sort v₁} {δ : Sort v₂}
+variable (R : α → β → Prop) (S : γ → δ → Prop)
+
+/-- The binary relations `R : α → β → Prop` and `S : γ → δ → Prop` induce a binary
+    relation on functions `lift_fun : (f : α → γ) (g : β → δ) : Prop'. -/
+def lift_fun (f : α → γ) (g : β → δ) : Prop :=
+∀⦃a b⦄, R a b → S (f a) (g b)
+
+/-- `(R ⇒ S) f g` means `lift_fun R S f g`. -/
+infixr:40 " ⇒ " => lift_fun
```
Comment (dwrensha) [inline Mathlib/Logic/Relator.lean:32]: I chose 40 for the precedence here because that's what Mathport chooses.

File: Mathlib/Logic/Relator.lean [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+-/
+
+section
+
+variable {α : Sort u₁} {β : Sort u₂} {γ : Sort v₁} {δ : Sort v₂}
+variable (R : α → β → Prop) (S : γ → δ → Prop)
+
+/-- The binary relations `R : α → β → Prop` and `S : γ → δ → Prop` induce a binary
+    relation on functions `lift_fun : (f : α → γ) (g : β → δ) : Prop'. -/
+def lift_fun (f : α → γ) (g : β → δ) : Prop :=
```
Comment (dwrensha) [inline Mathlib/Logic/Relator.lean:28]: Mathport makes all of these defs CamelCase (e.g. `LiftFun`), but it's unclear to me whether we've settled on a convention for this kind of thing.

Comment (dwrensha) [reply Mathlib/Logic/Relator.lean]: After an offline discussion with @digama0, I've renamed all of these to match the output of synport.

**Comment by semorrison** [comment]
bors merge

**Comment by semorrison** [comment]
Let's try that again.

bors merge

**Pull request closed without merging**